app/components/avo/index/resource_table_component.rb

```diff
@@ -15,11 +15,23 @@ def initialize(resources: nil, resource: nil, reflection: nil, parent_model: nil
   end
 
   def encrypted_query
-    return if @query.nil?
+    return if query.nil?
 
     Avo::Services::EncryptionService.encrypt(
-      message: @query.all.to_sql,
+      message: message_to_encrypt,
       purpose: :select_all
     )
   end
+
+  def message_to_encrypt
```

Comment:
I think we're being too clever here, and will bite us in the behind in the future. Why fallback to `.all`? Isn't that going to output all the items in a custom adapter?

I was thinking of having a `serialized` method on the adapter. The `ActiveRecord` one would use `to_sql` and others will implement their own.

How about we just check if it responds to `to_sql`. It it doesn't, then output nothing, and let's disable the feature in the UI.
How does that sound?

Reply:
We can do it. First approach was to struggle and send it anyway, that way users with custom HTTP resources could use the select all feature. But i agree it can be quite risky, if the object to encrypt is to large it can break.